tests/unit/setup-container-test-test.js

```diff
@@ -78,4 +79,70 @@ describe('setupContainerTest', function() {
       this.set('name', 'blue');
     });
   });
+
+  describe('hooks API', function() {
+
+    describe('calls hooks in order', function() {
+      function setupFoo(hooks) {
+        hooks.beforeEach(function() {
+          expect(this.owner, 'Context is set up already').to.exist;
+          this.count = 0;
```

Comment:
- due to the context resetting stuff we do we should probably track this outside of `this`
- I think it would be better to track the order with a step list instead of incrementing a counter